### PR TITLE
add cmake option DISABLE_UPDATE_CHECKER 

### PR DESCRIPTION
**Pull request by Alaskra**
close https://github.com/flameshot-org/flameshot/issues/1706 .
allow to disable update checker for the packaged builds with a special compile-time flag: `-DDISABLE_UPDATE_CHECKER:BOOL=ON`.

### CODE REVIEW

**Comment by Alaskra** [comment]
I run `clang-format -i $(git ls-files "*.cpp" "*.h")` and it changes a bunch of files...

**Comment by Alaskra** [comment]
I can't figure out clang-format, the `SplitEmptyRecord` option seems don't take effort. I fallback the wrong formatting manually.

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,6 +68,10 @@ option(GENERATE_TS "Regenerate translation source files" OFF)
 option(USE_EXTERNAL_SINGLEAPPLICATION "Use external QtSingleApplication library" OFF)
 option(USE_LAUNCHER_ABSOLUTE_PATH "Use absolute path for the desktop launcher" ON)
 option(USE_WAYLAND_CLIPBOARD "USE KF Gui Wayland Clipboard" OFF)
+option(DISABLE_UPDATE_CHECKER "Disable check for updates" OFF)
+if (DISABLE_UPDATE_CHECKER)
+  add_definitions(-DDISABLE_UPDATE_CHECKER="ON")
```
Comment (borgmanJeremy) [inline CMakeLists.txt:73]: This is not needed. You can reference USE_WAYLAND_CLIPBOARD

**Review by borgmanJeremy — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (Alaskra) [reply CMakeLists.txt:73]: But it add definition in here:
https://github.com/flameshot-org/flameshot/blob/57c8e8d893669cd83d8b279ea3ff6db50b16aa48/src/CMakeLists.txt#L208-L211
Do I need to change to this way?

Comment (Alaskra) [reply CMakeLists.txt]: And I find `add_compile_definitions` is more recommend.

**Pull request merged**